kernel/src/history_manager/mod.rs

```diff
@@ -2,3 +2,5 @@
 pub mod search;
 #[cfg(not(feature = "internal-api"))]
 pub(crate) mod search;
+
+mod timestamp_visitor;
```

Comment:
organization nit: if this is a generally-useful ICT visitor, do you think we should instead include it in the existing visitors module? (or do a sub-module?)

Reply:
I could move it to the visitors module 👍